internal/scan.go

```diff
@@ -0,0 +1,242 @@
+	TotalCount       int
+	AffectedRepos    int
+	VulnsByEcosystem map[string]int
+	VulnsBySeverity  map[string]int
+}
+
+func NewVulnerabilityReport() vulnerabilityReport {
+	return vulnerabilityReport{
+		AffectedRepos:    0,
+		TotalCount:       0,
+		VulnsBySeverity:  NewSeverityMap(),
+		VulnsByEcosystem: map[string]int{},
+	}
+}
+
+func collateSummaryReport(repos []api.VulnerabilityRepository) (report vulnerabilityReport) {
+	log := logger.Get()
+	report = NewVulnerabilityReport()
+	for _, repo := range repos {
+		repoVulns := repo.VulnerabilityAlerts.TotalCount
+		report.TotalCount += repoVulns
+		if repoVulns > 0 {
+			report.AffectedRepos += 1
+		}
+		tallyVulnsBySeverity(repo.VulnerabilityAlerts.Nodes, report.VulnsBySeverity)
+		tallyVulnsByEcosystem(repo.VulnerabilityAlerts.Nodes, report.VulnsByEcosystem)
+	}
+	log.Debug().Any("report", report).Msg("Collated summary report.")
+	return report
+}
+
+func groupVulnsByOwner(repos []api.VulnerabilityRepository, owners map[string][]string) map[string][]api.VulnerabilityRepository {
+	vulnsByTeam := map[string][]api.VulnerabilityRepository{}
+	// First, group up the repositories by owner
+	for _, repo := range repos {
+		owners := getRepositoryOwners(repo.Name, owners)
+		if len(owners) == 0 {
+			owners = []string{NO_OWNER_KEY}
+		}
+		for _, slug := range owners {
+			_, exists := vulnsByTeam[slug]
+			if !exists {
+				vulnsByTeam[slug] = make([]api.VulnerabilityRepository, 0)
+			}
+			vulnsByTeam[slug] = append(vulnsByTeam[slug], repo)
+		}
+	}
+	return vulnsByTeam
+}
+
+func collateTeamReports(vulnsByTeam map[string][]api.VulnerabilityRepository) (teamReports map[string]map[string]vulnerabilityReport) {
+	log := logger.Get()
+
+	teamReports = map[string]map[string]vulnerabilityReport{}
+	for team, repos := range vulnsByTeam {
+		_, exists := teamReports[team]
+		if !exists {
+			teamReports[team] = map[string]vulnerabilityReport{}
+		}
+		teamReports[team][SUMMARY_KEY] = NewVulnerabilityReport()
+		for _, repo := range repos {
+			summaryReport, _ := teamReports[team][SUMMARY_KEY]
+			summaryReport.AffectedRepos += 1
+			repoReport := NewVulnerabilityReport()
+			tallyVulnsByEcosystem(repo.VulnerabilityAlerts.Nodes, repoReport.VulnsByEcosystem)
+			tallyVulnsByEcosystem(repo.VulnerabilityAlerts.Nodes, summaryReport.VulnsByEcosystem)
+			tallyVulnsBySeverity(repo.VulnerabilityAlerts.Nodes, repoReport.VulnsBySeverity)
+			for severity, count := range repoReport.VulnsBySeverity {
+				summaryReport.VulnsBySeverity[severity] += count
+				summaryReport.TotalCount += count
+				repoReport.TotalCount += count
+			}
+			teamReports[team][SUMMARY_KEY] = summaryReport
+			teamReports[team][repo.Name] = repoReport
+		}
+		log.Debug().Str("team", team).Any("teamReport", teamReports[team]).Msg("Completed team report.")
+	}
+	return teamReports
+}
+
+func Scan(cmd *cobra.Command, args []string) {
+	log := logger.Get()
+
+	// Load the configuration file
+	config := configs.LoadConfig()
```

Comment:
Since the `vulnbot/configs` module got renamed to `vulnbot/config`, I think this may cause a conflict. And we will probably need to rename some variables here. Maybe we use `appConfig` as the local variable? Or.... `userConfig`? I like that one.

Reply:
Yup! This just happened to me locally, I like `appConfig`. Will update

Reply:
Fixed!